specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2021-03-01/redisenterprise.json

```diff
@@ -0,0 +1,1643 @@
+            "modelAsString": true
+          }
+        },
+        "persistence": {
+          "$ref": "#/definitions/Persistence",
+          "description": "Persistence settings"
+        },
+        "modules": {
+          "type": "array",
+          "description": "Optional set of redis modules to enable in this database - modules can only be added at creation time.",
+          "items": {
+            "$ref": "#/definitions/Module"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      }
+    },
+    "DatabaseList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Database"
+          },
+          "description": "List of databases"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The URI to fetch the next page of results."
+        }
+      },
+      "description": "The response of a list-all operation."
+    },
+    "Persistence": {
+      "title": "Persistence settings",
+      "description": "Persistence-related configuration for the RedisEnterprise database",
+      "properties": {
+        "aofEnabled": {
+          "type": "boolean",
+          "description": "Sets whether AOF is enabled."
+        },
+        "rdbEnabled": {
+          "type": "boolean",
+          "description": "Sets whether RDB is enabled."
+        },
+        "aofFrequency": {
+          "type": "string",
+          "description": "Sets the frequency at which data is fsynced to disk.",
+          "enum": [
+            "1s",
+            "always"
+          ],
+          "x-ms-enum": {
+            "name": "AofFrequency",
+            "modelAsString": true
+          }
+        },
+        "rdbFrequency": {
+          "type": "string",
+          "description": "Sets the frequency at which a snapshot of the database is created.",
+          "enum": [
+            "1h",
+            "6h",
+            "12h"
+          ],
+          "x-ms-enum": {
+            "name": "RdbFrequency",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Module": {
+      "title": "Module settings",
+      "description": "Specifies configuration of a redis module",
+      "properties": {
+        "name": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "The name of the module, e.g. 'RedisBloom', 'RediSearch', 'RedisTimeSeries'"
+        },
+        "args": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "Configuration options for the module, e.g. 'ERROR_RATE 0.00 INITIAL_SIZE 400'."
+        },
+        "version": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The version of the module, e.g. '1.0'."
```

Comment:
I have a worry about this being in the REST API that we aren't really able to honor the intention going forward, since there are only certain set of versions we support at any time. It might be better to remove and leave implicitly 'latest version'?

Reply:
Never mind I forgot its readonly!